desktop/main/sources/currentNetwork.ts

```diff
@@ -1,14 +1,38 @@
 import { find } from 'ramda';
-import { combineLatest, map, Subject } from 'rxjs';
+import {
+  combineLatestWith,
+  from,
+  iif,
+  map,
+  Observable,
+  Subject,
+  switchMap,
+} from 'rxjs';
 import { Network, Wallet } from '../../../shared/types';
+import NodeConfig from '../NodeConfig';
 
 export default (
+  $runNodeBeforeLogin: Observable<boolean>,
   $wallet: Subject<Wallet | null>,
   $networks: Subject<Network[]>
 ) =>
-  combineLatest([$wallet, $networks]).pipe(
+  $runNodeBeforeLogin.pipe(
+    switchMap((runNodeBeforeLogin) =>
+      iif(
+        () => runNodeBeforeLogin,
```

Comment:
Seems like a place for a potential bug. What if...
1. I set up autostart (so `runNodeBeforeLogin` is `true`)
2. Then I restarted Smapp, and it started smeshing on the current network (let's say 224)
3. Then I unlock the wallet and switch network to another one (225)
4. But since the flag is still `true` and we don't watch for NodeConfig (just load it once) so the `$currentNetwork` value might be outdated.

What I think we can do here:
1. If the flag is `false` or `$wallet` is `null`, then use the `$nodeConfig` subject (instead of loading it once)
2. Otherwise use `$wallet`

Reply:
Changed it as u described.